From da9f89ae70219f357fba6fffd2dae1ec886d8a3b Mon Sep 17 00:00:00 2001
From: deviance <meangczac.chan@gmail.com>
Date: Wed, 10 Mar 2021 18:12:25 +0800
Subject: [PATCH] Prevent unwanted vars redirection. Just to mitigate flooding
 of error logs, redirect if someone do malicious attempts on the search
 script.

---
 .../PHPFusion/Search/Search_Engine.php        | 35 +++++++++++--------
 1 file changed, 20 insertions(+), 15 deletions(-)

diff --git a/includes/classes/PHPFusion/Search/Search_Engine.php b/includes/classes/PHPFusion/Search/Search_Engine.php
index 0ad12eac99..c5320abd51 100644
--- a/includes/classes/PHPFusion/Search/Search_Engine.php
+++ b/includes/classes/PHPFusion/Search/Search_Engine.php
@@ -295,21 +295,26 @@ protected static function display_search_form() {
      * @return array|string
      */
     public static function get_param($key = NULL) {
-        $info = [
-            'stype'        => htmlentities(self::$search_type),
-            'stext'        => htmlentities(self::$search_text),
-            'method'       => htmlentities(self::$search_method),
-            'datelimit'    => self::$search_date_limit,
-            'fields'       => self::$search_fields,
-            'sort'         => self::$search_sort,
-            'chars'        => htmlentities(self::$search_chars),
-            'order'        => self::$search_order,
-            'forum_id'     => self::$forum_id,
-            'memory_limit' => self::$memory_limit,
-            'composevars'  => self::$composevars,
-            'rowstart'     => self::$rowstart,
-            'search_param' => self::$search_param,
-        ];
+        try {
+            $info = [
+                'stype'        => stripinput(self::$search_type),
+                'stext'        => stripinput(self::$search_text),
+                'method'       => stripinput(self::$search_method),
+                'datelimit'    => self::$search_date_limit,
+                'fields'       => self::$search_fields,
+                'sort'         => self::$search_sort,
+                'chars'        => stripinput(self::$search_chars),
+                'order'        => self::$search_order,
+                'forum_id'     => self::$forum_id,
+                'memory_limit' => self::$memory_limit,
+                'composevars'  => self::$composevars,
+                'rowstart'     => self::$rowstart,
+                'search_param' => self::$search_param,
+            ];
+        } catch (\Exception $e) {
+            redirect(BASEDIR.fusion_get_settings('opening_page'));
+        }
+
 
         return $key === NULL ? $info : (isset($info[$key]) ? $info[$key] : NULL);
     }
